app/Policies/SitePolicy.php

```diff
@@ -34,7 +34,17 @@ public function view(User $user, Site $site)
             return true;
         }
 
-        if($user->isFriendsWith($site->user)) {
+        if ($site->groups()->whereIn('group_id', $user->groups()
+            ->wherePivot('can_view', 1)->get()
+            ->pluck('pivot.group_id'))->exists()) {
```

Comment:
Is there a problem with simply plucking `id` here?
```suggestion
            ->pluck('id'))->exists()) {
```